gear-foundation/integrations-rosetta-api
# [transactions] Parse related operations for TransactionFeePaid event

## Description

Transaction fees are represented by three events:
1. A `Withdraw` event for the fee payer and the amount paid
2. A `Deposit` event for the fee recipient and the amount paid
3. A `TransactionFeePaid` event for the fee payer and the amount paid

In order to more clearly distinguish what the fee was, this change does the following:
1. Searches for the `TransactionFeePaid` event as the "debit" operation
2. Skips the first corresponding `Withdraw` event we see (i.e. the `Withdraw` event with a matching balance and address)

## Testing

Steps to Reproduce:
1. Change the base_path in `client/src/ApiClient.js` to `https://dev-archive-staking.vara-network.io`
5. Update `test/vara-test/config.json` and set `data/end_conditions/reconciliation_coverage/from_tip` to false
6. Run `yarn:build && start:vara-test`
7. Run the `check:data` command:
> $ rm -rf cli-data && ./bin/rosetta-cli check:data --configuration-file test/vara-test/config.json --start-block 1215000 --end-block 1216000

```
2023/06/28 11:23:34 Finished syncing 1215000-1216000
[MEMORY] Heap: 1134.424522MB Stack: 3.031250MB System: 1262.882118MB GCs: 6
2023/06/28 11:23:34 check:data status server shutting down
Index: 1216000

Success: Index End Condition [Index: 1216000]

+--------------------+--------------------------------+--------+
|  CHECK:DATA TESTS  |          DESCRIPTION           | STATUS |
+--------------------+--------------------------------+--------+
| Request/Response   | Rosetta implementation         | PASSED |
|                    | serviced all requests          |        |
+--------------------+--------------------------------+--------+
| Response Assertion | All responses are correctly    | PASSED |
|                    | formatted                      |        |
+--------------------+--------------------------------+--------+
| Block Syncing      | Blocks are connected into a    | PASSED |
|                    | single canonical chain         |        |
+--------------------+--------------------------------+--------+
| Balance Tracking   | Account balances did not go    | PASSED |
|                    | negative                       |        |
+--------------------+--------------------------------+--------+
| Reconciliation     | No balance discrepancies were  | PASSED |
|                    | found between computed and     |        |
|                    | live balances                  |        |
+--------------------+--------------------------------+--------+

+--------------------------+--------------------------------+-------------+
|     CHECK:DATA STATS     |          DESCRIPTION           |    VALUE    |
+--------------------------+--------------------------------+-------------+
| Blocks                   | # of blocks synced             |        1001 |
+--------------------------+--------------------------------+-------------+
| Orphans                  | # of blocks orphaned           |           0 |
+--------------------------+--------------------------------+-------------+
| Transactions             | # of transaction processed     |           1 |
+--------------------------+--------------------------------+-------------+
| Operations               | # of operations processed      |           4 |
+--------------------------+--------------------------------+-------------+
| Accounts                 | # of accounts seen             |           3 |
+--------------------------+--------------------------------+-------------+
| Active Reconciliations   | # of reconciliations performed |           3 |
|                          | after seeing an account in a   |             |
|                          | block                          |             |
+--------------------------+--------------------------------+-------------+
| Inactive Reconciliations | # of reconciliations performed |           3 |
|                          | on randomly selected accounts  |             |
+--------------------------+--------------------------------+-------------+
| Exempt Reconciliations   | # of reconciliation failures   |           0 |
|                          | considered exempt              |             |
+--------------------------+--------------------------------+-------------+
| Failed Reconciliations   | # of reconciliation failures   |           0 |
+--------------------------+--------------------------------+-------------+
| Skipped Reconciliations  | # of reconciliations skipped   |           0 |
+--------------------------+--------------------------------+-------------+
| Reconciliation Coverage  | % of accounts that have been   | 100.000000% |
|                          | reconciled                     |             |
+--------------------------+--------------------------------+-------------+

badger 2023/06/28 11:23:35 INFO: Storing value log head: {Fid:0 Len:32 Offset:748343}
````
